src/mscorlib/src/System/Text/UnicodeEncoding.cs

```diff
@@ -1444,8 +1466,10 @@ internal override unsafe int GetCharCount(byte* bytes, int count, DecoderNLS bas
                             fallbackBuffer.InternalInitialize(byteStart, charEnd);
                         }
 
-                        if (!fallbackBuffer.InternalFallback(byteBuffer, bytes, ref chars))
+                        charsForFallback = chars; // Avoid passing chars by reference to allow it to be enregistered
+                        if (!fallbackBuffer.InternalFallback(byteBuffer, bytes, ref charsForFallback))
                         {
+                            chars = charsForFallback;
```

Comment:
It might be clearer instead to set chars in only one place ie
```c#
                        charsForFallback = chars; // Avoid passing chars by reference to allow it to be enregistered
                         bool result = fallbackBuffer.InternalFallback(byteBuffer, bytes, ref charsForFallback);
                             chars = charsForFallback;
                         if (result)
                          {		                          {